src/coreclr/jit/gentree.cpp

```diff
@@ -8128,9 +8129,21 @@ GenTreeCall* Compiler::gtCloneExprCallHelper(GenTreeCall* tree, unsigned addFlag
     // a shallow copy suffices.
     copy->tailCallInfo = tree->tailCallInfo;
 
-    copy->gtCallType    = tree->gtCallType;
-    copy->gtReturnType  = tree->gtReturnType;
-    copy->gtControlExpr = gtCloneExpr(tree->gtControlExpr, addFlags, deepVarNum, deepVarVal);
+    copy->gtCallType   = tree->gtCallType;
+    copy->gtReturnType = tree->gtReturnType;
+
+    // Are we in fgGlobalMorph and did we just expand this virtual method call target early?
+    //
+    if (fgGlobalMorph && tree->IsExpandedEarly() && tree->IsVirtualVtable())
+    {
+        // The 'copy' node will create its own expanded sequence when we morph this cloned node
```

Comment:
Presumably `gtControlExpr` is `nullptr` here? Could we check for that instead?

Reply:
This method gets called from  fgMorphCopyBlock (and several other places)

And when called from fgMorphCopyBlock the gtControlExpr has already been expanded as we start from the bottom of the tree and morph upwards.  It also could be nullptr if we are planning on expanding this callsite late.

Reply:
I guess I don't understand why this needs to be dependent on `fgGlobalMorph` and can't simply be
```
    if (tree->gtControlExpr == nullptr)
```

Reply:
Because for some calls we want to expand Virtual Calls late in Lower and for the others we will expand them earlier during Morph. We could revisit this decision, but originally it was pretty simple to implement this way, and I wanted to be able to turn off this change and revert back to the old way to allow us to investigate any bugs that might come up due to this change.
 
By checking `(tree->gtControlExpr == nullptr)` this would match all the cases where we want to expand the call late in Lower, so instaed we always need to check `tree->IsExpandedEarly() && tree->IsVirtualVtable()`. 

Here I also have to check `fgGlobalMorph`, since morph is called multiple times when we optimize (in CSE, etc..), only during the first "global morph" phase do we want to expand the gtControlExpr.  

On these last couple of issues that I am fixing, the problem is that for some unusual code paths, we weren't calling  gtMorphTree at all while in the "global morph" phase.
Thus the `gtControlExpr` was not expanded and the call flags were set to tell Lower that it should not expand `gtControlExpr` as it should have been expanded early.

Reply:
Oh Joy,
Now when I add the call to fgMorpgTree of the cloned tree for `STRESS_CLONE_EXPR `it now fails for asmall number of methods with a  `'!"Missing flags on tree"'` assertion cause by `optAssertionProp_Ind` setting the GTF_ORDER_SIDEEFF flag and not propagating it up the tree.

This only happens for a couple of methods when JIitStress is on.

Reply:
Sorry harp on this, but are you saying if we get here and we're in global morph and are cloning an expand early call and `gtControlExpr` is non-null, we don't clone the existing tree because we know we'll create a proper new tree shortly..?

Won't that just create a tree that's more or less identical to the one we'd get it we did clone the existing tree?

Reply:
I will investigate doing it that way.

Reply:
That is what we had previously and it causes the assert in the Issue #48403
In he field by field copy path we call fgMorphTree on a cloned tree.

I will try a different fix.